### ISSUE TITLE
plejd-ble: warning: wasn't able to connect to Plejd

### ISSUE THREAD

**Issue by prebbz** [icanos/hassio-plejd#110]
Hi,
Running hass.io on an Intel NUC with BLE. After configuring the plugin, the following is printed in an endless loop:

``` parse error: Expected string key before ':' at line 1, column 4
[19:33:22] ERROR: Unknown HTTP error occured
 System:   (amd64 / qemux86-64)
 Home Assistant version: 0.117.1
 Supervisor version: 2020.10.0
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub, forums or the Discord chat.
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[19:33:22] INFO: Starting the Plejd service...
[19:33:22] INFO: Verified permissions on startup script
[19:33:22] INFO: Executing startup script
[19:33:23] INFO: Wrote plejd.json
[19:33:23] INFO: Running add-on
starting Plejd add-on v. 0.4.7
plejd-api: login()
plejd-api: logging into xxxxx
plejd-api: got session token response
plejd-api: getSites()
plejd-api: got site list response
plejd-api: getSite(...)
plejd-api: got site details response
wireEvents()
init()
plejd-mqtt: connected to mqtt.
plejd-mqtt: discovered light (DIM-01) named Entrance window with PID 11.
plejd-mqtt: discovered light (DIM-01) named Avlastningsbord with PID 13.
plejd-mqtt: discovered light (DIM-01) named Entrance with PID 15.
plejd-mqtt: discovered light (DIM-01) named Kitchen island with PID 17.
plejd-mqtt: discovered light (DIM-01) named Dining table  with PID 18.
plejd-mqtt: discovered light (DIM-01) named Stairs with PID 16.
plejd-mqtt: discovered light (DIM-02) named Mood light  with PID 19.
plejd-mqtt: discovered light (DIM-02) named Living room overhead with PID 20.
onDeviceConnected()
plejd-ble: warning: wasn't able to connect to Plejd, will retry.
plejd-ble: were unable to connect, will retry connection in 10 seconds.
init()
onDeviceConnected()
plejd-ble: warning: wasn't able to connect to Plejd, will retry.
plejd-ble: were unable to connect, will retry connection in 10 seconds.
init()
onDeviceConnected()
plejd-ble: warning: wasn't able to connect to Plejd, will retry.
plejd-ble: were unable to connect, will retry connection in 10 seconds.
init()
onDeviceConnected()
```

On each "scan", the output from btmon goes crazy with activity, so it definitely seems like some form of BLE is happening. Grateful for any tips.

Mvh
/Marcus

**Comment by f00dagi** [comment]
It is probably not very helpful but I am running this integration on a NUC with BLE without any issues. Are you sure you filled out the configuration part correctly?

**Comment by foogy77** [comment]
I have the same output in my log.

**Comment by prebbz** [comment]
Hi all, thanks for the responses. They were more helpful than you might think.
I have filled in the configuration except for the mqtt section to configuration.yaml (I didn't think it was necessary for testing, and the error message doesn't indicate an MQTT error)

**Comment by f00dagi** [comment]
So your problem is solved? The integration works really well, been using it for more than a year. The only thing I am missing is full support for transition and brightness.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by prebbz** [comment]
No, the problem is far from solved. Did you require the config in configuration.yaml for this to work?

Skickat från min iPhone

**Comment by f00dagi** [comment]
Have you installed the mqtt add-on and added the mqtt: part in the configuration.yaml. In the Plajd add-on configuration you need to point at your mqtt installation.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by foogy77** [comment]
I have mosquitto running on a different host. I guess that should be ok? Or is mqtt broker or client need to be added to hassio?

**Comment by prebbz** [comment]
Hi again all, 
- I have the MQTT add-on, and I added a plejd user to it 
- I have now added the MQTT section to my HA's configuration.yaml pointing to localhost and that user
- The MQTT plugin logs shows the plugin connecting

```
onDeviceConnected()
plejd-ble: warning: wasn't able to connect to Plejd, will retry.
plejd-ble: were unable to connect, will retry connection in 10 seconds.
```

It seems that at least one other person has this issue - @icanos  do you have any pointers?

**Comment by foogy77** [comment]
As the log message indicates (plejd-ble), I suspect Bluetooth connection problems. 
Can the connection be tested manually without using this plugin?

**Comment by f00dagi** [comment]
https://hass-io.blogspot.com/2019/12/hardware-add-plejd-to-hassio.html?m=1&fbclid=IwAR1dNQCt1bku3TTDdmUVtg7dbAGzVthaeyYJX-0J33VB5rN9kEQMpc3Nylk

I wrote this blogpost some time back, in Swedish though, it might help?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by prebbz** [comment]
Bra post! Dock var det 100% på engelska :) och dessutom har jag följt alla dessa instruktioner så har ingen aning vad är felet.

Skickat från min iPhone

**Comment by prebbz** [comment]
I'm happy to try another BLE adapter, is there any known-to-work device I should purchase?

**Comment by foogy77** [comment]
It is working for me now.
I did two things
1. Upgraded my server from Ubuntu 16.04 to 18.04. (Blutooth is now working better)
2. Added bluetooth_le_tracker according to https://www.home-assistant.io/integrations/bluetooth_le_tracker/

**Comment by foogy77** [comment]
> I'm happy to try another BLE adapter, is there any known-to-work device I should purchase?

I'm using this adapter https://www.kjell.com/se/produkter/mobilt/bluetooth/bluetooth-adaptrar/bluetooth-usb-adaptrar/plexgear-bluetooth-adapter-med-aptx-p92954

**Comment by zissou1** [comment]
Im using this integration without any problems, hassio on rpi4. However, I’m planning on transferring to a NUC. What NUC are you using this integration with which works out-of-the-box? Plejd is a central part in my home automation and I want it to work without hassle of external Bluetooth adapters. Thanks.

**Comment by prebbz** [comment]
Hi @zissou1  I purchased the Kjell & company adapter that @foogy77 mentions and the problems remain even after disabling the inbuilt device. The only thing left for me to upgrade to 18.04 (I have 16.04 but running a lot of other crap on here such as pihole, DHCP webserver etc) so i am reluctant to upgrade until I have time to fix any problems

**Comment by f00dagi** [comment]
>Im using this integration without any problems, hassio on rpi4. However, I’m planning on transferring to a NUC. What NUC are you using this integration with which works out-of-the-box? Plejd is a central part in my home automation and I want it to work without hassle of external Bluetooth adapters. Thanks.

@zissou1 I did that rpi4 to nuc transfer three months ago and I have a nuc7i5bnk with plejd. I am using the built in bluetooth on ubuntu 18 with supervised home assistant installation. As I understand it you must have bluetooth ble (and therefore bluetooth 4.0 or later)? I did not get it to work with the hassOS image.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by zissou1** [comment]
Thank you! Valuable info. I was planning on running HassOS but gues I have to rethink.

**Comment by SweVictor** [comment]
@zissou1 : Not sure if this is still relevant or helpful, but I do have it running under Windows => VirtualBox => HomeAssistant image with supervisor and everything => USB dongle, so it IS possible

**Comment by zissou1** [comment]
Thanks. I recieved my NUC and installed the HassOS image for NUC directly on the disc and BT works great. No I worried for nothing:)

**Comment by SweVictor** [comment]
@prebbz : I suppose you still have issues with this? The new version adds a lot more logging if you set logLevel to verbose or even silly, so maybe you could try that?

I have also added a bit to the readme regarding what helped me ("Having issues" heading, not merged to master yet)
https://github.com/icanos/hassio-plejd/blob/be2fbf34e8e294728f65e0690a8bf1aa667fe408/plejd/README.md not sure if it helps. Unfortunately I'm not the right person to guide on Bluetooth issues in Linux...

**Comment by prebbz** [comment]
@SweVictor  thanks heaps for following up. 
I still have the same issues - but I really do think they will be solved by updating to Ubuntu 18.04 (or higher) which will improve the bluetooth stack. The reason for not doing this yet is I have a VPN server, MySQL, pihole along heaps of cron jobs as well as HA so have been reluctant to tackle this until I have time to properly fix things if the upgrade goes wrong.

**Comment by SweVictor** [comment]
Ok, thanks for that update. I'll close this now to reduce "old" issues. Feel free to re-open or open a new issue if the problems persist after new Linux version.
